packages/dev/core/src/Materials/meshDebugPluginMaterial.ts

```diff
@@ -0,0 +1,358 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import type { Engine } from "../Engines/engine";
+import { MaterialPluginBase } from "./materialPluginBase";
+import type { Scene } from "../scene";
+import type { UniformBuffer } from "./uniformBuffer";
+import type { Nullable } from "../types";
+import { MaterialDefines } from "./materialDefines";
+import type { PBRBaseMaterial } from "./PBR/pbrBaseMaterial";
+import type { StandardMaterial } from "./standardMaterial";
+import type { SubMesh } from "../Meshes/subMesh";
+import { RegisterClass } from "../Misc/typeStore";
+import { Color3, Vector3 } from "core/Maths/math";
+
+/**
+ * Supported visualizations of MeshDebugPluginMaterial
+ */
+export enum MeshDebugMode {
+    /**
+     * Material without any mesh debug visualization
+     */
+    NONE = 0,
+    /**
+     * Wireframe displayed over mesh
+     */
+    TRIANGLES = 1,
+    /**
+     * TODO: This is NOT IMPLEMENTED yet!

```

Comment:
So far, VERTICES has only been implemented outside of the material plugin. Will revisit this.